source/MissionAction.cpp

```diff
@@ -195,14 +222,41 @@ void MissionAction::Load(const DataNode &node, const string &missionName)
 			{
 				// outfit <outfit> 0 means the player must have this outfit.
 				child.PrintTrace("Warning: deprecated use of \"outfit\" with count of 0. Use \"require <outfit>\" instead:");
-				requiredOutfits[GameData::Outfits().Get(child.Token(1))] = 1;
+				OutfitLocations location;
+				requiredOutfits[GameData::Outfits().Get(child.Token(1))] = make_pair(1,location);
 			}
 		}
 		else if(key == "require" && hasValue)
 		{
 			int count = (child.Size() < 3 ? 1 : static_cast<int>(child.Value(2)));
+			
+			OutfitLocations location;
+			for(const DataNode &it : child)
+			{
+				location.empty = false;
+				for(int i = 0; i < it.Size(); ++i)
```

Comment:
This should be a range-based for loop.

Reply:
How would I do that? Looking around, normal for loops are used for traversing the tokens of a DataNode in a similar fashion in [NPC.cpp](https://github.com/endless-sky/endless-sky/blob/07dd814209ed1e1ccf9547fd7fa86fa340097cf8/source/NPC.cpp#L49) and [Interface.cpp](https://github.com/endless-sky/endless-sky/blob/6dc743e3217756e2d031a231af05dc8a8e06076f/source/Interface.cpp#L41).

Reply:
In #3128  I extend `DataNode` to expose the tokens vector. Without that, the begin/end iterators invoked by the range-based for loop would be accessing child `DataNode`s, rather than the string tokens.

Reply:
If this is the currently-accepted paradigm, then it's fine until/unless the aforementioned PR gets merged.